src/frontend/components/Chat/SharedChatComponents/InputDisplayNameOverlay/index.spec.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { renderImageSnapshot } from 'utils/tests/imageSnapshot';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { converse } from 'utils/window';
+import { InputDisplayNameOverlay } from './index';
+
+const mockSetInputBarActive = jest.fn();
+const mockSetOverlay = jest.fn();
+
+jest.mock('utils/window', () => ({
+  converse: {
+    claimNewNicknameInChatRoom: jest.fn(),
+  },
+}));
+
+describe('<InputDisplayNameOverlay />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("controls input and show error when input contains 'Anonymous'", () => {
+    render(
+      wrapInIntlProvider(
+        <InputDisplayNameOverlay
+          setInputBarActive={mockSetInputBarActive}
+          setOverlay={mockSetOverlay}
+        />,
+      ),
+    );
+    const inputTextbox = screen.getByRole('textbox');
+    const validateButton = screen.getByRole('button');
+    userEvent.type(inputTextbox, 'Anonymous-John');
+    act(() => userEvent.click(validateButton));
+    screen.getByText('Keyword "anonymous" is not allowed.');
+    expect(screen.queryByText('Min lenght is 3 characters.')).toBeNull();
+    expect(screen.queryByText('Max lenght is 15 characters.')).toBeNull();
+    expect(inputTextbox).toHaveValue('Anonymous-John');
+    expect(mockSetInputBarActive).not.toHaveBeenCalled();
+    expect(mockSetOverlay).not.toHaveBeenCalled();
+  });
+
+  it('controls input and show error when input contains less than 3 characters.', () => {
+    render(
+      wrapInIntlProvider(
+        <InputDisplayNameOverlay
+          setInputBarActive={mockSetInputBarActive}
+          setOverlay={mockSetOverlay}
+        />,
+      ),
+    );
+    const inputTextbox = screen.getByRole('textbox');
+    const validateButton = screen.getByRole('button');
+    userEvent.type(inputTextbox, 'JD');
+    act(() => userEvent.click(validateButton));
+    screen.getByText('Min lenght is 3 characters.');
+    expect(
+      screen.queryByText('Keyword "anonymous" is not allowed.'),
+    ).toBeNull();
+    expect(screen.queryByText('Max lenght is 15 characters.')).toBeNull();
+    expect(inputTextbox).toHaveValue('JD');
+    expect(mockSetInputBarActive).not.toHaveBeenCalled();
+    expect(mockSetOverlay).not.toHaveBeenCalled();
+  });
+
+  it('controls input and show error when input contains more than 15 characters.', () => {
```

Comment:
```suggestion
  it('controls input and shows error when input contains more than 15 characters.', () => {
```

Reply:
done in commit [d3c2a0](https://github.com/openfun/marsha/pull/1276/commits/d3c2a037d1f2b70d6fd332d3c8951c3bea0df3a5)